code/workspaces/auth-service/src/userManagement/authZeroUserManagement.js

```diff
@@ -18,30 +20,57 @@ const authKeyMapping = {
   user_id: 'userId',
 };
 
-export function asyncGetUsers() {
-  return requestAccessToken(accessTokenRequest())
-    .then(bearer => axios.get(`${authZeroManagementApi}/users`, {
-      params: {
-        fields: 'name,user_id',
-        per_page: '100',
-      },
-      headers: {
-        Authorization: `Bearer ${bearer}`,
-      },
-    }).then(extractUsers)
-      .then(processUsers)
-      .catch(() => {
-        throw new Error('Unable to retrieve users from User Management Service.');
-      }))
-    .catch((err) => {
+export async function asyncGetUsers() {
+  try {
+    try {
+      let page = 0;
+      let users = [];
+      let extractedUsers = [];
+      let activeUsers = [];
+      let allUsers = [];
+
+      const tokenRequest = await accessTokenRequest();
+      const bearer = await requestAccessToken(tokenRequest);
+
+      // Below approach is required as the auth0 API results are paginated, with a maximum
+      // overall limit, as well as a maximum per_page limit. This approach (non-async) is
+      // taken to prevent excessive API calls which impact performance.
+      /* eslint-disable no-await-in-loop */
+      do {
+        users = await fetchUsers(bearer, page);
+        extractedUsers = await extractUsers(users);
+        activeUsers = extractedUsers.filter(user => user.name);
+        allUsers = allUsers.concat(activeUsers);
+        page += 1;
+      } while (extractedUsers.length === PAGE_LIMIT && page < (AUTH0_MAXIMUM / PAGE_LIMIT));
+      /* eslint-enable no-await-in-loop */
+
+      const processedUsers = await processUsers(allUsers);
+      return processedUsers;
+    } catch (err) {
       logger.error(err.message);
-      throw err;
-    });
+      throw new Error('Unable to retrieve users from User Management Service.');
+    }
+  } catch (err) {
+    logger.error(err.message);
+    throw err;
+  }
 }
 
+const fetchUsers = (bearer, page) => axios.get(`${authZeroManagementApi}/users`, {
+  params: {
+    fields: 'name,user_id',
+    per_page: PAGE_LIMIT,
+    page,
+  },
+  headers: {
+    Authorization: `Bearer ${bearer}`,
+  },
+});
+
 // Pending users are users with an account but have not logged in; these users do not have a populated name field.
 // This function filters pending users and returns only active users.
```

Comment:
If you leave this function as not containing the filter then you will need to update/remove this comment